build,refactor: update development tooling and package configuration (#614)
C2SM/icon4py: CODE_OF_CONDUCT.md

# Contributor Covenant Code of Conduct

## Our Pledge

> We as members, contributors, and leaders pledge to make participation in our
> community a harassment-free experience for everyone, regardless of age, body

Reviewer: ```suggestion
As members, contributors, and leaders, we pledge to contribute to create a harassment-free environment within our community for everyone, regardless of age, body
```

Author: This code of conduct is the latest version from the widely adopted Contributor Covenant (https://www.contributor-covenant.org/), which I found to be a reasonable semi-standard code of conduct, so just added it to the repo and almost forgot about it.

I now think the proper way to add a code of conduct would be to discuss it in the team first, and then open a new PR to add whatever the agreed version, instead of sneaking it here, so I'll remove it from this PR.

size, visible or invisible disability, ethnicity, sex characteristics, gender
identity and expression, level of experience, education, socio-economic status,
nationality, personal appearance, race, caste, color, religion, or sexual
identity and orientation.

We pledge to act and interact in ways that contribute to an open, welcoming,
diverse, inclusive, and healthy community.

## Our Standards

Examples of behavior that contributes to a positive environment for our
community include:

* Demonstrating empathy and kindness toward other people
* Being respectful of differing opinions, viewpoints, and experiences
* Giving and gracefully accepting constructive feedback
* Accepting responsibility and apologizing to those affected by our mistakes,
  and learning from the experience

> * Focusing on what is best not just for us as individuals, but for the overall

Reviewer: ```suggestion
* Focusing on what is best not only for ourselves as individuals, but for the overall
```

  community

Examples of unacceptable behavior include:

* The use of sexualized language or imagery, and sexual attention or advances of
  any kind
* Trolling, insulting or derogatory comments, and personal or political attacks
* Public or private harassment
* Publishing others' private information, such as a physical or email address,
  without their explicit permission
* Other conduct which could reasonably be considered inappropriate in a

>   professional setting

Reviewer: This is not specific enough to be categorized as an example

## Enforcement Responsibilities

Community leaders are responsible for clarifying and enforcing our standards of

> acceptable behavior and will take appropriate and fair corrective action in

Reviewer: ```suggestion
acceptable behavior and will take appropriate, fair, and corrective actions in
```

response to any behavior that they deem inappropriate, threatening, offensive,
or harmful.

Community leaders have the right and responsibility to remove, edit, or reject

> comments, commits, code, wiki edits, issues, and other contributions that are
> not aligned to this Code of Conduct, and will communicate reasons for moderation

Reviewer: ```suggestion
comments, commits, code, wiki edits, issues, and other contributions that do
not align with this Code of Conduct, and will communicate reasons for moderation
```

decisions when appropriate.

## Scope

> This Code of Conduct applies within all community spaces, and also applies when

Reviewer: ```suggestion
This Code of Conduct applies within all community spaces as well as when
```

an individual is officially representing the community in public spaces.

> Examples of representing our community include using an official email address,

Reviewer: ```suggestion
Examples of representing our community include: using an official email address,
```

posting via an official social media account, or acting as an appointed
representative at an online or offline event.

## Enforcement

Instances of abusive, harassing, or otherwise unacceptable behavior may be
reported to the community leaders responsible for enforcement at
[magdalena.luz@c2sm.ethz.ch](mailto:magdalena.luz@c2sm.ethz.ch).
All complaints will be reviewed and investigated promptly and fairly.

All community leaders are obligated to respect the privacy and security of the
reporter of any incident.

## Enforcement Guidelines

Community leaders will follow these Community Impact Guidelines in determining

> the consequences for any action they deem in violation of this Code of Conduct:

Reviewer: ```suggestion
the consequences for any action they deem in violation of this Code of Conduct.
```

### 1. Correction

**Community Impact**: Use of inappropriate language or other behavior deemed
unprofessional or unwelcome in the community.

**Consequence**: A private, written warning from community leaders, providing
clarity around the nature of the violation and an explanation of why the
behavior was inappropriate. A public apology may be requested.

### 2. Warning

**Community Impact**: A violation through a single incident or series of
actions.

**Consequence**: A warning with consequences for continued behavior. No
interaction with the people involved, including unsolicited interaction with
those enforcing the Code of Conduct, for a specified period of time. This
includes avoiding interactions in community spaces as well as external channels
like social media. Violating these terms may lead to a temporary or permanent
ban.

### 3. Temporary Ban

**Community Impact**: A serious violation of community standards, including
sustained inappropriate behavior.

**Consequence**: A temporary ban from any sort of interaction or public
communication with the community for a specified period of time. No public or

> private interaction with the people involved, including unsolicited interaction
> with those enforcing the Code of Conduct, is allowed during this period.

Reviewer: ```suggestion
private interaction with the people involved (including unsolicited interaction
with those enforcing the Code of Conduct) is allowed during this period.
```
